modules/media/php/media.class.inc

```diff
@@ -118,12 +124,22 @@ class Media extends \NDB_Menu_Filter
                );
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @return \Loris\Data\Provisioner
+     */
+    public function getBaseDataProvisioner(): \LORIS\Data\Provisioner
+    {
+        return new MediaFileProvisioner();
+    }
+
     /**
      * Gets the data source for this menu filter.
      *
      * @return \LORIS\Data\Provisioner
      */
-    function getDataProvisioner() : \LORIS\Data\Provisioner
+    public function getDataProvisionerWithFilters() : \LORIS\Data\Provisioner
     {
         $provisioner = new MediaFileProvisioner();
```

Comment:
```
        $provisioner = parent::getDataProvisionerWithFilters();
        
        if (!$user->hasPermission("superuser")) {
            $provisioner = $provisioner->filter(
                new HideFileFilter()
            );
        }
        
        return $provisioner;
```

Reply:
@johnsaigle This suggestion was applied to the wrong place. 
`getBaseDataProvisioner` should have 1 line `return  new MediaFileProvisioner();`

And `getDataProvisionerWithFilters` body should be :
```
        $provisioner = parent::getDataProvisionerWithFilters();
        
        if (!\NDB_Factory::singleton()->user()->hasPermission("superuser")) {
            $provisioner = $provisioner->filter(
                new HideFileFilter()
            );
        }
        
        return $provisioner;
```

The `parent::getDataProvisionerWithFilters();` will take care of 'access_all_profiles'.

Reply:
Thanks, I also just realized this after looking at the Travis logs.